shared/js/initApp.js

```diff
@@ -150,7 +150,35 @@ $(window).on('function_editor_closed', function() {
   }
 })(appOptions.level);
 
+dashboard.saveProject = function(callback) {
+  var app_id = dashboard.currentApp.id;
+  dashboard.currentApp.startBlocks = Blockly.Xml.domToText(Blockly.Xml.blockSpaceToDom(Blockly.mainBlockSpace));
+  if (app_id) {
+    storageApps().update(app_id, dashboard.currentApp, function(data) {
+      callback(data);
+    });
+  } else {
+    storageApps().create(dashboard.currentApp, function(data) {
+      if (history) {
+        history.pushState({}, '', '?id=' + data.id);
+      }
+      callback(data);
+    });
+  }
+};
+
 function initApp() {
+  if (appOptions.level.isProject) {
+    if (dashboard.currentApp) {
+      appOptions.level.startBlocks = dashboard.currentApp.startBlocks;
+    } else {
+      dashboard.currentApp = {
```

Comment:
you will have to store a level name (or some other way to get back to the level), right? Or are you doing that somewhere that I missed?